tools/generate_config.py

```diff
@@ -0,0 +1,326 @@
+                source.write('%s},\n' % indent)
+                scope = scope[:-1]
+          else:
+            scope.append(name)
+            sname = '.'.join(scope)
+            ctype = pre_type(jtype) + post_type(jtype)
+            dflt = jtype['default'] if 'default' in jtype else ''
+            if dflt == '':
+              raise Exception("ERROR: no default provided for %s" % sname)
+            if ctype == 'bool':
+              dflt = "true" if dflt else "false"
+            if ctype == 'char*' or ctype.startswith('char['):
+              if dflt is None or dflt == []:
+                source.write('%s.%s=NULL,\n' % (indent, name))
+              else:
+                source.write('%s.%s="%s",\n' % (indent, name, dflt))
+            else:
+              if need_size_var(jtype):
+                size_var_name = 'num_' + name;
+                if name == 'key':
+                  size_var_name = 'key_len'
+                source.write('%s.%s=%s,\n' % (indent, size_var_name, 0))
+              if dflt is None or dflt == []:
+                dflt = 'NULL'
+              source.write('%s.%s=%s,\n' % (indent, name, dflt))
+          scope = scope[:-1]
+      scope = scope[:-1]
+    initialize(scope, 'sgxlkl_enclave_config_t')
+    source.write('};\n\n')
+
+    source.write('// clang-format off\n')
+    source.write('const sgxlkl_enclave_setting_t sgxlkl_enclave_settings[%s] = {\n' % num_settings)
+    scope = []
+    def describe(scope, elem):
+      typedef = root['definitions'][elem]
+      if 'enum' not in typedef:
+        for name, jtype in typedef['properties'].items():
+          if name == 'format_version':
+            continue
+          tname = pre_type(jtype)
+          if (tname.endswith('*')):
+            tname = tname[:-1]
+          if tname.startswith('sgxlkl_'):
+            scope.append(name)
+            describe(scope, tname)
+            scope = scope[:-1]
+          else:
+            scope.append(name)
+            sname = '.'.join(scope)
+            ctype = pre_type(jtype) + post_type(jtype)
+            desc = jtype['description'] if 'description' in jtype else ''
+            desc = desc.replace("\"", "\\\"")
+            dflt = 'NULL'
+            if 'default' in jtype:
+              dflt = jtype['default'];
+            if dflt == [] or dflt == None:
+                dflt = 'NULL'
+            if ctype == 'bool':
+              dflt = "true" if dflt else "false"
+            override_var = 'NULL'
+            if 'overridable' in jtype:
+                override_var = '"' + jtype['overridable'] + '"'
+            source.write('    {"%s", "%s", "%s", "%s", %s},\n' % (sname, ctype, desc, dflt, override_var))
+          scope = scope[:-1]
+      scope = scope[:-1]
+    describe(scope, 'sgxlkl_enclave_config_t')
+
+    source.write('};\n')
+    source.write('// clang-format on\n')
+
+def generate(args):
+  schema_file_name = str(ENCLAVE_CFG_SCHEMA_PATH)
+  with open(schema_file_name, "r") as schema_file:
```

Comment:
`"r"` is the default.

Reply:
So? I'd rather know about it immediately instead of going off wondering and trying to find the Python docs for the defaults of my particular flavor of Python...

Reply:
"r" is the default in every Python version and implementation. Not an issue.